tests/specs/infinite-gas.k

```diff
@@ -0,0 +1,55 @@
+requires "evm.md"
+
+module INFINITE-GAS
+    imports EVM
+
+
+ // Symbolic Gas
+
+    syntax Int ::= #gas ( Int , Int , Int )  [function]  // startGas, nonMemory, memory
+ //-----------------------------------------------------------------------------------
+
+ //
+ // Infinite Gas
+ // Here we use the construct `#gas` to count _up_ gas used instead of counting down.
+ // This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
+ //
+
+    syntax Int ::= #gas ( Int ) [function, functional, no-evaluators]
+ // -----------------------------------------------------------------
+    rule #gas(_, _, _) => #gas(0)
+
+    rule #gas(G) -Int G' => #gas(G +Int G') [simplification]
+
+    rule #gas(_)  <Int I => false requires I <=Int pow256 [simplification]
```

Comment:
@ehildenb Why not this?

```suggestion
    rule #gas(_)  <Int I => notBool (I <=Int pow256) [simplification]
```

Reply:
@ttuegel we don't want this change because it's not always the case that we can simplify out `#gas(_) <Int I` using the fact that `I <=Int pow256`, it's just a heuristic. So basically, `I <=Int pow256 IMPLIES notBool #gas(_) <Int I`, but _not_ the case that `I <=Int pow256 EQUALS notBool (I <=Int pow256)`.

For example, if you have `#gas(_) <Int (2 *Int pow256)`, and you simplify it to `notBool (2 *Int pow256 <=Int pow256)`, you'll conclude `true`, when instead it should take the rule `rule #gas(_) <Int I => false [concrete(_I)]` to determine `true`.

The reason for this rule is that many values which are compared in EVM are bounded by `pow256`, so this is a good heuristic for simplifying out using the fact that `#gas(_)` is larger than any particular number.

Reply:
> instead it should take the rule `rule #gas(_) <Int I => false [concrete(_I)]` to determine `true`.

Do you mean `false`?

Reply:
Yes, sorry, edited.